src/rulesDeploy.ts

```diff
@@ -100,6 +109,48 @@ export class RulesDeploy {
     return { latestName, latestContent };
   }
 
+  async checkStorageRulesIamPermissions(rulesContent?: string): Promise<void> {
+    if (rulesContent?.match(CROSS_SERVICE_FUNCTIONS) === null) {
+      return;
+    }
+
+    // We have cross-service rules. Now check the P4SA permission
+    const projectNumber = await getProjectNumber(this.options);
+    const saEmail = `service-${projectNumber}@gcp-sa-firebasestorage.iam.gserviceaccount.com`;
+    try {
+      if (await serviceAccountHasRoles(projectNumber, saEmail, [CROSS_SERVICE_RULES_ROLE], true)) {
+        return;
+      }
+
+      // Prompt user to ask if they want to add the service account
+      const addRole = await promptOnce(
```

Comment:
You _might_ want to check if `this.options.nonInteractive` is true - for someone in a CI/CD environment a prompt has the chance of stalling out the command.